src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -30,9 +31,233 @@ namespace {
 
 namespace query = xrt_core::query;
 using key_type = query::key_type;
+xclDeviceHandle handle;
+xclDeviceInfo2 deviceInfo;
 
 static std::map<query::key_type, std::unique_ptr<query::request>> query_tbl;
 
+static zynq_device*
+get_edgedev(const xrt_core::device* device)
+{
+  return zynq_device::get_dev();
+}
+
+struct devInfo
+{
+   static boost::any
+    get(const xrt_core::device* device,key_type key)
+  {
+    auto edev = get_edgedev(device);
+    switch (key)
+    {
+    case key_type::edge_device:
+      return deviceInfo.mDeviceId;
+    case key_type::edge_vendor:
+      return deviceInfo.mVendorId;
+    case key_type::edge_subsystem_vendor:
+      return deviceInfo.mSubsystemVendorId;
+    case key_type::edge_subsystem_id:
+      return deviceInfo.mSubsystemId;
+    case key_type::rom_vbnv:
+      return std::string(deviceInfo.mName);
+    case key_type::rom_ddr_bank_size:
+      return deviceInfo.mDDRSize;
+    case key_type::rom_ddr_bank_count_max:
+      return deviceInfo.mDDRBankCount;
+    /*returning dummy value for registered pcie_bdf query request*/
+    case key_type::pcie_bdf:
+      {
+        uint16_t ret = std::numeric_limits<uint16_t>::max();
+        return std::make_tuple(ret, ret, ret);
+      }
+    default:
+      return std::string("NA");
+    }
+  }
+};
+
+// Specialize for other value types.
+template <typename ValueType>
+struct sysfs_fcn
+{
+  static ValueType
+    get(zynq_device* dev, const char* entry)
+  {
+    std::string err;
+    ValueType value;
+    dev->sysfs_get(entry, err, value, static_cast<ValueType>(-1));
```

Comment:
For errors, exceptions should be thrown and should be caught by a try catch block.